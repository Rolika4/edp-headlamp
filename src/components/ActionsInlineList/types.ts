import { KubeObjectAction } from '../../types/actions';

export interface ActionsInlineListProps {
  actions: KubeObjectAction[];
}
