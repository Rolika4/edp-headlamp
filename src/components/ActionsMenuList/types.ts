import { KubeObjectAction } from '../../types/actions';

export interface ActionsMenuListProps {
  actions: KubeObjectAction[];
  handleCloseActionsMenu?(): void;
  anchorEl: HTMLElement | null;
}
