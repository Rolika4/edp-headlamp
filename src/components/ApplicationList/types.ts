import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';

export interface ApplicationListProps {
    applications: EDPCodebaseKubeObjectInterface[];
    onEdit?(): void;
    onDelete?(): void;
}
