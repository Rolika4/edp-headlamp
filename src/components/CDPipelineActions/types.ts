import { EDPCDPipelineKubeObjectInterface } from '../../k8s/EDPCDPipeline/types';

export interface CDPipelineActionsProps {
    kubeObjectData: EDPCDPipelineKubeObjectInterface;
    isDetailsPage?: boolean;
}
