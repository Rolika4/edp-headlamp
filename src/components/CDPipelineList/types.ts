import { EDPCDPipelineKubeObjectInterface } from '../../k8s/EDPCDPipeline/types';

export interface CDPipelineListProps {
    CDPipelines: EDPCDPipelineKubeObjectInterface[];
}
