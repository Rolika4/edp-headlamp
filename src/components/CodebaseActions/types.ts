import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';

export interface CodebaseActionsProps {
    kubeObjectData: EDPCodebaseKubeObjectInterface;
    isDetailsPage?: boolean;
}
