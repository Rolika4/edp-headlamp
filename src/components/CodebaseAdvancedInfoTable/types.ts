import { EDPKubeObjectInterface } from '../../types/k8s';

export interface CodebaseAdvancedInfoTableProps {
    kubeObjectData: EDPKubeObjectInterface;
}
