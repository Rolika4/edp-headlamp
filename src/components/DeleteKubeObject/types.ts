import { React } from '../../plugin.globals';
import type { k8s } from '../../plugin.types';
import { DeepPartial } from '../../types/global';
import { EDPKubeObjectInterface } from '../../types/k8s';

export interface DeleteKubeObjectProps {
    popupOpen: boolean;
    setPopupOpen(boolean): void;
    kubeObject: k8s.cluster.KubeObject;
    kubeObjectData: DeepPartial<EDPKubeObjectInterface>;
    objectName: string;
    description: string;
    onBeforeSubmit?(
        setErrorTemplate: React.Dispatch<React.SetStateAction<React.ReactNode>>,
        setLoadingActive: React.Dispatch<React.SetStateAction<boolean>>
    ): Promise<void>;
    onSuccess?: () => void;
}
