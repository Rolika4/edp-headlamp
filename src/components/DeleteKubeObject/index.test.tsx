/**
 * @jest-environment jsdom
 */

import { describe, test } from '@jest/globals';
import { configureStore } from '@reduxjs/toolkit';
import { render, screen } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';
import React from 'react';
import { Provider } from 'react-redux';
import { DeleteKubeObject } from './index';
import { DeleteKubeObjectProps } from './types';

describe('DeleteKubeObject', () => {
    test('renders correctly', () => {
        const props: DeleteKubeObjectProps = {
            popupOpen: true,
            setPopupOpen: () => {},
            kubeObject: null,
            kubeObjectData: {
                apiVersion: 'apiextensions.k8s.io/v1',
                kind: 'CustomResourceDefinition',
                metadata: {
                    name: 'TestCustomResourceDefinition',
                    namespace: 'best-namespace-ever',
                    finalizers: [],
                    generation: 1,
                    uid: '',
                    creationTimestamp: new Date().toISOString(),
                    resourceVersion: 'unknown',
                    selfLink: '',
                },
            },
            objectName: 'super-cool-object-name',
            description: 'Confirmation message for test action',
            onBeforeSubmit: () => Promise.resolve(),
        };
        const store = configureStore({
            reducer: () => ({}),
        });

        render(
            <Provider store={store}>
                <SnackbarProvider>
                    <DeleteKubeObject {...props} />
                </SnackbarProvider>
            </Provider>
        );

        expect(screen.getByRole('dialog')).toMatchSnapshot();
    });
});
