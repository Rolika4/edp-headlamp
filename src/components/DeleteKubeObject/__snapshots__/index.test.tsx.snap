// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteKubeObject renders correctly 1`] = `
<div
  class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
  role="dialog"
>
  <div
    class="MuiDialogTitle-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h6"
    >
      Confirm deletion of "super-cool-object-name"
    </h2>
  </div>
  <div
    class="MuiDialogContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Confirmation message for test action
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <form>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Enter CustomResourceDefinition name to delete
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="name"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-3"
                    >
                      <span>
                        Enter CustomResourceDefinition name to delete
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiDialogActions-root MuiDialogActions-spacing"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Cancel
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="submit"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Confirm
                  </span>
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
